state/volume.go

```diff
@@ -266,7 +275,19 @@ func (st *State) addVolumeOp(params VolumeParams) (txn.Op, names.VolumeTag, erro
 	if err := st.validateVolumeParams(params); err != nil {
 		return txn.Op{}, names.VolumeTag{}, errors.Annotate(err, "validating volume params")
 	}
-	name, err := newVolumeName(st)
+	_, provider, err := poolStorageProvider(st, params.Pool)
```

Comment:
I've moved the scope validation code to storage.go, updated via validate{Volume,Filesystem}Params.

If the scope is Environ and machineId != "", we should *not* error. We're creating a volume for a machine, but the volume doesn't have to be scoped to that machine.